pcsx2/USB/usb-pad/usb-mascon.h

```diff
@@ -0,0 +1,240 @@
+		/* bcdUSB              */ WBVAL(0x0110), /* USB 1.1 */ \
+		/* bDeviceClass        */ 0xFF, \
+		/* bDeviceSubClass     */ subclass, \
+		/* bDeviceProtocol     */ 0x00, \
+		/* bMaxPacketSize0     */ 0x08, \
+		/* idVendor            */ WBVAL(0x0ae4), \
+		/* idProduct           */ WBVAL(product), \
+		/* bcdDevice           */ WBVAL(0x0102), /* 1.02 */ \
+		/* iManufacturer       */ 0x01, \
+		/* iProduct            */ 0x02, \
+		/* iSerialNumber       */ 0x03, \
+		/* bNumConfigurations  */ 0x01, \
+	}
+
+	// These settings are common across multiple models.
+	static const uint8_t taito_denshacon_config_descriptor[] = {
+		USB_CONFIGURATION_DESC_SIZE, // bLength
+		USB_CONFIGURATION_DESCRIPTOR_TYPE, // bDescriptorType
+		WBVAL(25), // wTotalLength
+		0x01, // bNumInterfaces
+		0x01, // bConfigurationValue
+		0x00, // iConfiguration (String Index)
+		0xA0, // bmAttributes
+		0xFA, // bMaxPower 500mA
+
+		USB_INTERFACE_DESC_SIZE, // bLength
+		USB_INTERFACE_DESCRIPTOR_TYPE, // bDescriptorType
+		0x00, // bInterfaceNumber
+		0x00, // bAlternateSetting
+		0x01, // bNumEndpoints
+		USB_CLASS_HID, // bInterfaceClass
+		0x00, // bInterfaceSubClass
+		0x00, // bInterfaceProtocol
+		0x00, // iInterface (String Index)
+
+		USB_ENDPOINT_DESC_SIZE, // bLength
+		USB_ENDPOINT_DESCRIPTOR_TYPE, // bDescriptorType
+		USB_ENDPOINT_IN(1), // bEndpointAddress (IN/D2H)
+		USB_ENDPOINT_TYPE_INTERRUPT, // bmAttributes (Interrupt)
+		WBVAL(8), // wMaxPacketSize
+		0x14, // bInterval 20 (unit depends on device speed)
+		// 25 bytes (43 total with dev descriptor)
+	};
+
+#define MASCON_PREAMBLE \
+	0x05, 0x01, \
+		0x09, 0x04, \
+		0xA1, 0x01
+
+#define MASCON_PAD_BYTE \
+	0x75, 0x08, \
+		0x95, 0x01, \
+		0x81, 0x01
+
+#define MASCON_AXES \
+	0x09, 0x01, \
+		0xA1, 0x00, \
+		0x09, 0x30, \
+		0x09, 0x31, \
+		0x09, 0x32, \
+		0x15, 0x00, \
+		0x26, 0xFF, 0x00, \
+		0x75, 0x08, \
+		0x95, 0x03, \
+		0x81, 0x02, \
+		0xC0
+
+#define MASCON_DPAD \
+	0x05, 0x01, \
+		0x25, 0x07, \
+		0x46, 0x3B, 0x01, \
+		0x75, 0x04, \
+		0x95, 0x01, \
+		0x65, 0x14, \
+		0x09, 0x39, \
+		0x81, 0x42, \
+		0x65, 0x00, \
+		0x95, 0x01
+
+#define MASCON_BUTTONS(max) \
+	0x81, 0x01, \
+		0x05, 0x09, \
+		0x19, 0x01, \
+		0x29, max, \
+		0x15, 0x00, \
+		0x25, 0x01, \
+		0x35, 0x00, \
+		0x45, 0x01, \
+		0x75, 0x01, \
+		0x95, max, \
+		0x81, 0x02, \
+		0x95, 8 - max, \
+		0x81, 0x01
+
+#define MASCON_END 0xC0 // End Collection
+
+	// ---- Two handle controller "Type 2" ----
+
+	static const USBDescStrings dct01_desc_strings = {
+		"",
+		"TAITO",
+		"TAITO_DENSYA_CON_T01",
+		"TCPP20009",
+	};
+
+	DEFINE_DCT_DEV_DESCRIPTOR(dct01, 0x04, 0x0004);
+
+	// https://marcriera.github.io/ddgo-controller-docs/controllers/usb/descriptors/tcpp20009_hid-report-descriptor.txt
+	static const uint8_t dct01_hid_report_descriptor[] = {
+		MASCON_PREAMBLE,
+		MASCON_PAD_BYTE,
+		MASCON_AXES, // Brake, Power, Horn
+		MASCON_DPAD,
+		MASCON_BUTTONS(6),
+		MASCON_END,
+	};
```

Comment:
`*_hid_report_descriptor` and `#define MASCON_*` are unused. These controllers aren't HID class devices, but Vendor specific : bDeviceClass=0xff.